sgkit/io/vcfzarr_reader.py

```diff
@@ -42,35 +48,213 @@ def read_vcfzarr(path: PathType) -> xr.Dataset:
 
     vcfzarr = zarr.open_group(str(path), mode="r")
 
-    # Index the contig names
-    variants_chrom = da.from_zarr(vcfzarr["variants/CHROM"]).astype(str)
-    variant_contig, variant_contig_names = encode_array(variants_chrom.compute())
-    variant_contig = variant_contig.astype("int16")
-    variant_contig_names = list(variant_contig_names)
+    # don't fix strings since it requires a pass over the whole dataset
+    return _vcfzarr_to_dataset(vcfzarr, fix_strings=False)
+
+
+def vcfzarr_to_zarr(
+    input: PathType,
+    output: PathType,
+    *,
+    contigs: Optional[List[str]] = None,
+    grouped_by_contig: bool = False,
+    consolidated: bool = False,
+    tempdir: Optional[PathType] = None,
+    concat_algorithm: Optional[str] = None,
```

Comment:
you could probably use `Literal` type here to make it more user friendly and safe.